src/fides/api/service/connectors/query_config.py

```diff
@@ -810,14 +813,41 @@ class BigQueryQueryConfig(QueryStringWithoutTuplesOverrideQueryConfig):
     Generates SQL valid for BigQuery
     """
 
+    def __init__(
+        self,
+        node: ExecutionNode,
+        namespace_meta: Optional[BigQueryNamespaceMeta] = None,
+    ):
+        """
+        Accepts an optional namespace_meta param to be able to specify dataset and project IDs for the generated queries.
+        """
+        super().__init__(node)
+        self.namespace_meta = namespace_meta
+
+    def _generate_table_name(self) -> str:
+        """
+        Prepends the dataset ID and project ID to the base table name
+        if the BigQuery namespace meta is provided.
+        """
+
+        table_name = self.node.collection.name
+        if self.namespace_meta:
+            table_name = f"{self.namespace_meta.dataset_id}.{table_name}"
+            if project_id := self.namespace_meta.project_id:
+                table_name = f"{project_id}.{table_name}"
+        return table_name
+
     def get_formatted_query_string(
         self,
         field_list: str,
         clauses: List[str],
     ) -> str:
-        """Returns a query string with backtick formatting for tables that have the same names as
-        BigQuery reserved words."""
-        return f'SELECT {field_list} FROM `{self.node.collection.name}` WHERE {" OR ".join(clauses)}'
+        """
+        Returns a query string with backtick formatting for tables that have the same names as
+        BigQuery reserved words.
+        """
+
+        return f'SELECT {field_list} FROM `{self._generate_table_name()}` WHERE {" OR ".join(clauses)}'
```

Comment:
Here it is 😄! 

(Mel was over my shoulder as I typed here it is and suggested I add a smiley face)